src/library_pthread.js

```diff
@@ -479,9 +479,12 @@ var LibraryPThread = {
         + '\nIf you want to throw an explicit error instead of the risk of deadlocking in those cases, use setting `-s PTHREAD_POOL_SIZE_STRICT=2`.'
 #endif
         );
-#else
-        err('No available workers in the PThread pool');
-#endif
+#else // PTHREAD_POOL_SIZE_STRICT
+#if ASSERTIONS
```

Comment:
Seems like a duplication, the branch before `else` is also for `ASSERTIONS`.

Reply:
Thanks, all the ifdefs confused me and I matched the wrong ifs with elses and ends...

Reply:
I know the feeling. Another reason to switch to regular JS ifs :)